src/Microsoft.DotNet.Wpf/tests/UnitTests/PresentationCore.Tests/System/Windows/ClipboardTests.cs

```diff
@@ -0,0 +1,246 @@
+        Clipboard.ContainsFileDropList().Should().BeTrue();
+    }
+
+    [WpfFact]
+    public void SetFileDropList_NullFilePaths_ThrowsArgumentNullException()
+    {
+        Action action = () => Clipboard.SetFileDropList(null!);
+        // Note: The name will change with the WinForms shared code.
+        action.Should().Throw<ArgumentNullException>().WithParameterName("fileDropList");
+    }
+
+    [WpfFact]
+    public void SetFileDropList_EmptyFilePaths_ThrowsArgumentException()
+    {
+        Action action = static () => Clipboard.SetFileDropList([]);
+        action.Should().Throw<ArgumentException>();
+    }
+
+    [WpfTheory]
+    [InlineData("")]
+    [InlineData("\0")]
+    public void SetFileDropList_InvalidFileInPaths_ThrowsArgumentException(string filePath)
+    {
+        StringCollection filePaths =
+        [
+            filePath
+        ];
+
+        Action action = () => Clipboard.SetFileDropList(filePaths);
+        action.Should().Throw<ArgumentException>();
+    }
+
+    [WpfFact]
+    public unsafe void SetImage_InvokeBitmap_VerifyPixelColor()
+    {
+        WriteableBitmap bitmap = new(10, 10, 96, 96, PixelFormats.Bgra32, palette: null);
+
+        // Set a specific pixel to a given color (e.g., set pixel at (1, 2) to red)
+        Color color = Colors.Red;
+        byte[] colorData = [color.B, color.G, color.R, color.A];
+        bitmap.WritePixels(new Int32Rect(1, 2, 1, 1), colorData, 4, 0);
+
+        Clipboard.SetImage(bitmap);
+        Clipboard.ContainsImage().Should().BeTrue();
+        InteropBitmap result = Clipboard.GetImage().Should().BeOfType<InteropBitmap>().Subject;
+
+        // Verify the pixel color
+        byte[] resultColorData = new byte[4];
+        result.CopyPixels(new Int32Rect(1, 2, 1, 1), resultColorData, 4, 0);
+        resultColorData.Should().Equal(colorData);
+
+        // Set back the image we just got from the clipboard
+        Clipboard.SetImage(result);
+        Clipboard.ContainsImage().Should().BeTrue();
+        result = Clipboard.GetImage().Should().BeOfType<InteropBitmap>().Subject;
+
+        // Verify the pixel color
+        result.CopyPixels(new Int32Rect(1, 2, 1, 1), resultColorData, 4, 0);
+        resultColorData.Should().Equal(colorData);
+    }
+
+    [WpfTheory]
+    [BoolData]
+    public void SetDataObject_WithMultipleData(bool copy)
+    {
+        string testData1 = "test data one";
+        int testData2 = 42;
+        DataObject data = new();
+        data.SetData("testData1", testData1);
+        data.SetData("testData2", testData2);
+        Clipboard.SetDataObject(data, copy);
+
+        object? result1 = Clipboard.GetData("testData1");
+        result1.Should().Be(testData1);
+        object? result2 = Clipboard.GetData("testData2");
+        result2.Should().Be(testData2);
+    }
+
+    [WpfFact]
+    public void SetData_Text_Format_AllUpper()
+    {
+        // The fact that casing on input matters is likely incorrect, but behavior has been this way.
```

Comment:
how do the tests below show the case sensitivity?

Reply:
"Text" is the "right" name for text, you'll see that in the formats list below. The not-quite right case sensitivity is with our DataStore implementations.